src/settings/Token.js

```diff
@@ -7,26 +7,72 @@ import { stripesShape } from '@folio/stripes/core';
 import { ConfigForm } from '@folio/stripes/smart-components';
 import { Col, Row, TextField } from '@folio/stripes/components';
 
-
+/**
+ * Display decoded JWT token. Set a new token.
+ */
 class Token extends React.Component {
   static propTypes = {
     label: PropTypes.string.isRequired,
     stripes: stripesShape.isRequired,
   };
 
-  constructor() {
-    super();
-    this.onSave = this.onSave.bind(this);
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      raw: props.stripes.store.getState().okapi.token,
+      cooked: {},
+      error: '',
+    };
+  }
+
+  /**
+   * decode the original token
+   */
+  componentDidMount = () => {
+    this.setState((state) => ({
+      cooked: this.parseToken(state.raw),
+    }));
   }
 
-  onSave(data) {
```

Comment:
Nice to see the explicit `bind` going away `en passant`. Sheesh, what idiots we used to be back when we wrote code like that. Good thing we do everything exactly right these days.